tests/test_helper/parallelize.bash

```diff
@@ -0,0 +1,69 @@
+# This file contains set of helper functions to run processes or functions
+# in background and collect their results.
+#
+# Caveat: Output of the processes running in parallel may overlap
+#
+# Example usage:
+# function task_distributor() {
+#     # Reserve new process pool
+#     local process_pool=""
+#     process_pool=$(new_process_pool)
+#
+#     # Start multiple commands or functions on the background
+#     start_bg_task "$process_pool" function_1 argument_1 argument_2
+#     start_bg_task "$process_pool" command_2 "$ARG_FROM_VARIABLE"
+#
+#     # Wait for all tasks in the process pool to finish
+#     gather_bg_tasks "$process_pool"
+#     # If any of the tasks failed, this function will return with return code 1
+# }
+
+declare -g -A _PROCESS_POOL
+
+# new_process_pool
+#
+# Create new process pool and print its ID.
+function new_process_pool() {
```

Comment:
Bash functions share one global namespace, I'd prefer if we prefix function names with module name to make it apparent where they are defined and what they act on, at least for specialized functions like this.